internal/transport/transport_test.go

```diff
@@ -2450,3 +2452,20 @@ func TestConnectionError_Unwrap(t *testing.T) {
 		t.Error("ConnectionError does not unwrap")
 	}
 }
+
+// Verify Peer is set in server context.
+func TestPeerSetInServerContext(t *testing.T) {
+	server := setUpServerOnly(t, 0, &ServerConfig{}, suspended)
```

Comment:
`setUpServerOnly` does not setup any transports. So, `server.conns` in your `for` loop will be `nil`, and therefore this test will not be doing anything.

You either need to use `setUpWithOptions` or you need to explicitly do a `net.Dial()` to create a transport. Please see some other tests in the same file to get a better idea. Thanks.